### PR TITLE
add test spec for placement

### PR DESCRIPTION
**Pull request by SethBarberee**
Basic tests but its a start and tests `awful.placement`. Working on adding more but felt like I should PR this for right now.

### CODE REVIEW

**Comment by actionless** [comment]
have no clue why travis is failing on some of unrelated tests on one of the setups (https://travis-ci.com/awesomeWM/awesome/jobs/286708214) -- i was trying restarting it but result is the same

**Comment by Elv13** [comment]
> have no clue why travis is failing on some of unrelated tests on one of the setups (https://travis-ci.com/awesomeWM/awesome/jobs/286708214) -- i was trying restarting it but result is the same

It is the same one that make every other PR fails randomly since like September. I never spent that much time trying to fix it, but I could never reproduce it and the few shot in the dark I tried didn't change anything.

File: spec/awful/placement_spec.lua [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    package.loaded["awful.screen"] = {}
+
+    local place = require("awful.placement")
+
+    it("awful.placement.closest_corner (top left)", function()
+        local _, corner = place.closest_corner(
+        {coords=function() return {x = 100, y=100} end},
+        {include_sides = true, bounding_rect = {x=0, y=0, width=200, height=200}}
+        )
+        assert(corner == "top_left")
```
Comment (Elv13) [inline spec/awful/placement_spec.lua]: Eventually, if you add more, favor the likes of `assert.are_equal` which are part of the testing framework. It makes prettier output.

Comment (SethBarberee) [reply spec/awful/placement_spec.lua]: > Eventually, if you add more, favor the likes of `assert.are_equal` which are part of the testing framework. It makes prettier output.

Gotcha, will keep in mind for the future.

**Pull request merged**